inngest-core/src/main/kotlin/com/inngest/Environment.kt

```diff
@@ -10,4 +10,139 @@ object Environment {
             InngestHeaderKey.Framework.value to (framework?.value),
         ).filterValues { (it is String) }.entries.associate { (k, v) -> k to v!! }
     }
+
+    fun inngestAppId(
+        clientId: String,
+        serveId: String? = null,
+    ): String {
+        if (serveId != null) return serveId
+        return clientId
+    }
+
+    fun inngestEventKey(key: String? = null): String {
+        if (key != null) return key
+        return System.getenv(InngestSystem.EventKey.value) ?: ""
+    }
+
+    fun inngestSigningKey(
+        env: InngestEnv,
+        key: String? = null,
+    ): String {
+        if (key != null) return key
+
+        return when (env) {
+            InngestEnv.Dev -> "test"
+            else -> {
+                val signingKey = System.getenv(InngestSystem.SigningKey.value)
+                if (signingKey == null) {
+                    throw Exception("signing key is required")
+                }
+                signingKey
+            }
+        }
+    }
+
+    fun inngestEventApiBaseUrl(
+        env: String? = null,
+        url: String? = null,
+    ): String {
+        if (url != null) return url
+
+        val baseUrl = System.getenv(InngestSystem.EventApiBaseUrl.value)
+        if (baseUrl != null) {
+            return baseUrl
+        }
+
+        return when (inngestEnv(env)) {
+            InngestEnv.Dev -> "http://127.0.0.1:8288"
+            InngestEnv.Prod -> "https://inn.gs"
+            InngestEnv.Other -> "https://inn.gs"
+        }
+    }
+
+    fun inngestApiBaseUrl(
+        env: InngestEnv,
+        url: String? = null,
+    ): String {
+        if (url != null) return url
+
+        val baseUrl = System.getenv(InngestSystem.ApiBaseUrl.value)
+        if (baseUrl != null) {
+            return baseUrl
+        }
+
+        return when (env) {
+            InngestEnv.Dev -> "http://127.0.0.1:8288"
+            else -> "https://inngest"
+        }
+    }
+
+    fun inngestServeHost(host: String? = null): String? {
+        if (host != null) return host
+        return System.getenv(InngestSystem.ServeHost.value)
+    }
+
+    fun inngestServePath(path: String? = null): String? {
+        if (path != null) return path
+        return System.getenv(InngestSystem.ServePath.value)
+    }
+
+    fun inngestEnv(
+        env: String? = null,
+        isDev: Boolean? = null,
+    ): InngestEnv {
+        if (isDev != null) {
+            return when (isDev) {
+                true -> InngestEnv.Dev
+                false -> InngestEnv.Prod
+            }
+        }
+        val sysDev = System.getenv(InngestSystem.Dev.value)
+        if (sysDev != null) {
+            return when (sysDev) {
+                "0" -> InngestEnv.Prod
+                "1" -> InngestEnv.Dev
+                else -> {
+                    var other = InngestEnv.Other
+                    other.value = sysDev
```

Comment:
I'm not sure if `value` can be set to anything besides "other" here

Reply:
https://pl.kotl.in/MsCo3qptY

Reply:
hm I guess it's just an instance variable and nothing is stopping you from having separate enums with the same value or changing the value either. Never thought about that before 🤷

Reply:
yup, exactly. probably not the best way to use it, so we figure better ways in the future, we can swap it out.
The reason I know we want an `other` is because some users want to run integration tests in CI using our Dev Server but want to not make it in `dev` mode.